src/main/scala/edu/ie3/simona/model/participant2/ChargingHelper.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * © 2024. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.participant2
+
+import edu.ie3.util.scala.quantities.DefaultQuantities.{zeroKW, zeroKWh}
+import squants.{Dimensionless, Each, Energy, Power, Seconds}
+
+/** Provides some basic functionality that can be used wherever energy is stored
+  * within some kind of storage system.
+  */
+object ChargingHelper {
+
+  /** Calculates a new total amount of stored energy given an old amount of
+    * energy, a charging or discharging power and a time span. Minimum and
+    * maximum charging levels are adhered to.
+    *
+    * @param storedEnergy
+    *   The former amount of stored energy.
+    * @param power
+    *   The charging power that has been valid from startTick to endTick.
+    * @param startTick
+    *   The tick (in seconds) at which charging started.
+    * @param endTick
+    *   The tick (in seconds) at which charging ended.
+    * @param maxEnergy
+    *   The maximum allowed stored energy.
+    * @param minEnergy
+    *   The minimum allowed stored energy, default to 0 Wh.
+    * @param eta
+    *   The efficiency applied to charging and discharging.
+    * @return
+    *   The new amount of stored energy.
+    */
+  def calcEnergy(
+      storedEnergy: Energy,
+      power: Power,
+      startTick: Long,
+      endTick: Long,
+      maxEnergy: Energy,
+      minEnergy: Energy = zeroKWh,
+      eta: Dimensionless = Each(1),
+  ): Energy = {
+    val timespan = Seconds(endTick - startTick)
+    val energyChange = calcNetPower(power, eta) * timespan
+
+    // don't allow under- or overcharge e.g. due to tick rounding error
+    minEnergy.max(maxEnergy.min(storedEnergy + energyChange))
+  }
+
+  /** Calculates the next tick (if applicable) at which some target energy level
+    * has been reached.
+    *
+    * @param storedEnergy
+    *   The current amount of stored energy.
+    * @param power
+    *   The charging power.
+    * @param currentTick
+    *   The current tick.
+    * @param chargingEnergyTarget
+    *   The target energy when charging.
+    * @param dischargingEnergyTarget
+    *   The target energy when discharging.
+    * @param eta
+    *   The efficiency applied to charging and discharging.
+    * @param tolerance
+    *   Tolerance for zero charging power.
+    * @return
+    *   The tick at which the target is reached, if applicable.
+    */
+  def calcNextEventTick(
+      storedEnergy: Energy,
+      power: Power,
```

Comment:
Do we have a better naming, that makes it directly clearer that this is the power provided to or provided by inverter.

Reply:
To my knowledge it's called "gross power" but I found that term a bit gross. 😆